aarch64/src/mailbox.rs

```diff
@@ -13,23 +12,15 @@ const MBOX_WRITE: usize = 0x20;
 const MBOX_FULL: u32 = 0x8000_0000;
 const MBOX_EMPTY: u32 = 0x4000_0000;
 
-static MAILBOX: Lock<Option<&'static mut Mailbox>> = Lock::new("mailbox", None);
+static MAILBOX: Lock<Option<Mailbox>> = Lock::new("mailbox", None);
 
 /// Mailbox init.  Mainly initialises a lock to ensure only one mailbox request
 /// can be made at a time.  We have no heap at this point, so creating a mailbox
 /// that can be initialised based off the devicetree is rather convoluted.
 pub fn init(dt: &DeviceTree) {
     let node = LockNode::new();
     let mut mailbox = MAILBOX.lock(&node);
-    *mailbox = Some({
-        static MAYBE_MAILBOX: SyncUnsafeCell<MaybeUninit<Mailbox>> =
-            SyncUnsafeCell::new(MaybeUninit::uninit());
-        unsafe {
-            let maybe_mailbox = &mut *MAYBE_MAILBOX.get();
-            maybe_mailbox.write(Mailbox::new(dt, KZERO));
-            maybe_mailbox.assume_init_mut()
-        }
-    });
+    *mailbox = Some(Mailbox::new(dt, KZERO));
```

Comment:
This is so much simpler. Tbh, I have no idea why we had this complicated block.

Reply:
I'm not sure either, although the code to use it is fairly ugly:
```
   MAILBOX
         .lock(&node)
         .deref_mut()
         .as_mut()
         .map(|mb| {
             mb.request(&mut msg);
             let res = unsafe { msg.response };
             res.tags.body
         })
         .expect("mailbox not initialised")
```

Reply:
Simplified the code a little, so it's a little less fugly now...